### PR TITLE
Use a consistent size pointer in g_utf8_to_utf16

### PR DESCRIPTION
**Pull request by joshpeterson**
* The `glong` type can vary in size on some platforms (e.g. Windows * 64-bit).
* If `glong` is 4bytes and `gsize` is 8 bytes, then the high 4 bytes from g_utf8_to_utf16 will be uninitialized.
* Standardize on the `guint32` type, which is 4 bytes on all platforms.

### CODE REVIEW

File: mono/utils/strenc.h [hunk 1]
```diff
@@ -14,7 +14,7 @@
 #include <glib.h>
 #include <mono/utils/mono-publib.h>
 
-extern MONO_API gunichar2 *mono_unicode_from_external (const gchar *in, gsize *bytes);
+extern MONO_API gunichar2 *mono_unicode_from_external (const gchar *in, guint32 *bytes);
 extern MONO_API gchar *mono_utf8_from_external (const gchar *in);
```
Comment (vargaz) [inline mono/utils/strenc.h]: This is public api, so it cannot be changed.

Comment (joshpeterson) [reply mono/utils/strenc.h]: What options do we have then? I guess I could leave it as `gsize`, then initialize the value inside `mono_unicode_from_external`.

Comment (vargaz) [reply mono/utils/strenc.h]: Yes, unfortunately it needs to use a different local with a different type.

Comment (joshpeterson) [reply mono/utils/strenc.h]: Ok, I'll make that change then. Thanks!

Comment (joshpeterson) [reply mono/utils/strenc.h]: After some more investigation, the change to the implementation of `mono_unicode_from_external` is really all that we need to correct the problem we see. So I'll close this PR and open a new one with a smaller change.

**Pull request closed without merging**